gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dform_ttt2.lua

```diff
@@ -0,0 +1,560 @@
+		data.master:AddSlave(left)
+		data.master:AddSlave(right)
+		data.master:AddSlave(reset)
+	end
+
+	return right, left
+end
+
+---
+-- Adds a helpbox to the form
+-- @param table data The data for the helpbox
+-- @return @{Panel} The created helpbox
+-- @realm client
+function PANEL:MakeHelp(data)
+	local left = vgui.Create("DLabelTTT2", self)
+
+	left:SetText(data.label)
+	left:SetContentAlignment(7)
+	left:SetAutoStretchVertical(true)
+
+	left.paddingX = 10
+	left.paddingY = 5
+
+	left.Paint = function(slf, w, h)
+		derma.SkinHook("Paint", "HelpLabelTTT2", slf, w, h)
+
+		return true
+	end
+
+	-- make sure the height is based on the amount of text inside
+	left.PerformLayout = function(slf, w, h)
+		local textTranslated = LANG.TryTranslation(slf:GetText())
+
+		local textWrapped = draw.GetWrappedText(
+			textTranslated,
+			w - 2 * slf.paddingX,
+			slf:GetFont()
+		)
+		local _, heightText = draw.GetTextSize(textTranslated, slf:GetFont())
+
+		slf:SetSize(w, heightText * #textWrapped + 2 * slf.paddingY)
+	end
+
+	self:AddItem(left, nil)
+
+	left:InvalidateLayout(true)
+
+	return left
+end
+
+-- Adds a colormixer to the form
+-- @param table data The data for the colormixer
+-- @return @{Panel}, @{Panel} The created colormixer, The created label
+-- @realm client
+function PANEL:MakeColorMixer(data)
+	local left = vgui.Create("DLabelTTT2", self)
+	local right = vgui.Create("DPanel", self)
+
+	left:SetTall(data.height or 240)
+	right:SetTall(data.height or 240)
+
+	right:Dock(TOP)
+	right:DockPadding(10, 10, 10, 10)
+
+	left:SetText(data.label)
+
+
+
+	local colorMixer = vgui.Create("DColorMixer", right)
+
+	colorMixer:SetColor(data.initial or COLOR_WHITE)
+	colorMixer:SetAlphaBar(data.showAlphaBar or false)
+	colorMixer:SetPalette(data.showPalette or false)
+	colorMixer:Dock(FILL)
+
+	colorMixer.ValueChanged = function(slf, color)
+		if isfunction(data.onChange) then
+			data.onChange(slf, color)
+		end
+	end
+
+	left.Paint = function(slf, w, h)
+		derma.SkinHook("Paint", "FormLabelTTT2", slf, w, h)
+
+		return true
+	end
+
+	right.Paint = function(slf, w, h)
+		derma.SkinHook("Paint", "FormBoxTTT2", slf, w, h)
+
+		return true
+	end
+
+	right.SetEnabled = function(slf, enabled)
+		slf.m_bDisabled = not enabled
+
+		if enabled then
+			slf:SetMouseInputEnabled(true)
+		else
+			slf:SetMouseInputEnabled(false)
+		end
+
+		-- update colormixer as well
+		colorMixer:SetDisabled(not enabled)
+	end
+
+	self:AddItem(left, right)
+
+	if IsValid(data.master) and isfunction(data.master.AddSlave) then
+		data.master:AddSlave(left)
+		data.master:AddSlave(right)
+	end
+
+	return right, left
+end
+
+derma.DefineControl("DFormTTT2", "", PANEL, "DCollapsibleCategoryTTT2")
+
+-- SIMPLE WRAPPER FUNCTION
+
+---
+-- Creates a collapsable form used for menu sections
+-- @param @{Panel} parent The parent panel
+-- @param string name The name of the collapsable form,
+-- can be a language identifier
+-- @realm client
+function CreateForm(parent, name)
```

Comment:
Is it too generic? I quite like it, as it is a generic function

Reply:
It is not a generic form, it is a DFormTTT2 and should be renamed, to not block / collide with other addons naming/creating a global `CreateForm` function, also the name should indicate, that it will create a DFormTTT2 instead of a normal DForm to avoid confusion.

Reply:
Agreed